File: Go/45.JumpGameII.go
func jump(nums []int) int {
    min := make([]int, 10000)
    goal := len(nums)-1 
    if goal == 0 {
        return 0
    }
    for i, n := range(nums) {
        if i+n >= goal {
            return min[i]+1
        }
        for j:=i+1; j<=i+n; j++ {
            if min[j] == 0 || min[j] > min[i] + 1 {
                min[j] = min[i] + 1
            }
        }
    }
    return min[goal]
}

File: Python3/45.JumpGameII.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        goal = len(nums) - 1
        if goal == 0: return 0
        curReach, maxReach, step = 0, 0, 0
        for i, n in enumerate(nums[:goal]):
            maxReach = max(maxReach, i+n)
            if i == curReach:
                step += 1
                curReach = maxReach
                
        return step
